rpcs3/Input/basic_mouse_handler.cpp

```diff
@@ -89,11 +89,23 @@ void basic_mouse_handler::MouseScroll(QWheelEvent* event)
 	MouseHandlerBase::Scroll(event->angleDelta().y());
 }
 
+bool basic_mouse_handler::getMouseLockState()
+{
+	if (m_target)
+	{
+		auto mouseLocked = m_target->property("mouseLocked");
+		if (mouseLocked.isValid())
+			return mouseLocked.toBool();
+		return false;
+	}
+	return false;
+}
+
 void basic_mouse_handler::MouseMove(QMouseEvent* event)
 {
 	if (is_time_for_update())
 	{
-		if (m_target && m_target->visibility() == QWindow::Visibility::FullScreen && m_target->isActive())
+		if (m_target && m_target->isActive() && getMouseLockState())
 		{
```

Comment:
I wonder if this function is fast enough.
Also mouse lock != Full screen handling of the mouse

Reply:
I am a little worried on the speed aspect, but because of the limited class associations between the QWindow and the handler, it seemed like the path of least resistance.
Changing the type of m_target from QWindow -> gs_frame and propagating that through all of the various calls, or doing a hacky dynamic_cast<gs_frame> seemed like the only other ways I could think of to do this.

I'm open to other ideas however.

Re mouse lock != full screen, understood.  I'm hoping that this change would resolve complaints related to the full screen capture of the mouse in #7977 also.

Reply:
Why does the function have camelCase? That's reserved for Qt or cell functions and should never be used in rpcs3 internal code

Reply:
You're effectively breaking this codepath (and the one in the Keyboard handler) by using its code without being in full screen. The code here is only supposed to be used when the active window is in full screen mode and the mouse handler is active (or the mouse is used in the keyboard pad handler).

If you start using the code depending on some random flag you set in the settings, then the whole code does not make any sense.

Reply:
That code path just gets the delta movement from last movement, and sets the cursor back into the centre of the window.
Whether it's in fullscreen or windowed doesn't make much difference (apart from the ease of escape).

Those other conditions are to:
a. prevent use of m_target when it's not valid
b. prevent setting the cursor position when the game window is not actually the active window

The fullscreen situation before was only because if the cursor locking was also performed in windowed mode you'd be unable to leave the window at all, hence unable to turn off the locking, and unable to close the window.
As #7977 notes, this ability to turn on/off the cursor lock should have been in place instead of the fullscreen condition.
And it equally well applies against non-fullscreen mode as fullscreen mode.

Reply:
You might be right. I think I initially had this code full screen exclusive before rewriting it. Sorry for that. I'll test the PR tomorrow

Reply:
I've pushed the more hacky version with dynamic_cast.  This should allay any performance concerns, since we'd only be paying the dynamic_cast and data access, not going through the unordered_map type implementation of the QWindow properties.

Perhaps I'm misinterpreting the #7977 issue, but it sounds more like in a multi-monitor environment, when they have rpcs3 fullscreen (taking up an entire single monitor, they might still have numerous other monitors with non-games), they want to be able to access the mouse still for other applications.  The previous implementation did not allow this at all.  The mouse would always be locked to the game window.

Reply:
No. The mouse should only be locked if you actually use it in the mouse handler or the keyboard pad.
It shouldn't be an issue if you don't use a mouse.
You'd have to inquire if the user has these features enabled. Or just test if it is still locked even if you don't use any mouse input